packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx

```diff
@@ -139,12 +139,20 @@ export function DayPicker<TDate>(props: PickersCalendarProps<TDate>) {
       if (readOnly) {
         return;
       }
+
       // TODO possibly buggy line figure out and add tests
```

Comment:
Now that it's tested, the comment could be removed, or at least remove 'line' otherwise in a few months we will wonder why `let finalDate` can be buggy

```suggestion
      // TODO possibly buggy figure out and add tests
```

Reply:
I found the issue related to this comment ealier today https://github.com/mui/mui-x/issues/4703

I think to problem is that some date library represents dates as array so this check is problematic and we should pass an  `isRange` boolean instead to differentiate ranges and mono dates.

Reply:
Good to know 👍